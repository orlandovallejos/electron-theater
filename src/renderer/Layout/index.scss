.main {
  background: red;
  display: flex;
  height: 100vh;
}

.sidebar {
  background: green;
  width: 250px;
}

.content {
  background: blue;
  flex: 1;
}
