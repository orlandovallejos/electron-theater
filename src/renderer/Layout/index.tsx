/*------------------------------------------------*/
/* LIBRARIES
/*------------------------------------------------*/
import React, { Fragment } from 'react';
import { MemoryRouter as Router, Switch, Route } from 'react-router-dom';
/*------------------------------------------------*/
/* INTERNAL DEPENDENCIES
/*------------------------------------------------*/
// Components
import Home from '../Home';
import AppBar from '../common/AppBar';
import MovieView from '../MovieView';
// Styles
import LayoutWrapper, { ContentWrapper } from './style';

const Layout = () => (
  <Fragment>
    <AppBar />
    <LayoutWrapper>
      {/* <div className={styles.sidebar}>sidebar</div> */}
      <ContentWrapper>
        <Router>
          <Switch>
            <Route path="/movie/:movieId" component={MovieView} />
            <Route path="/serie/:serieId" component={MovieView} />
            <Route path="/" component={Home} />
          </Switch>
        </Router>
      </ContentWrapper>
    </LayoutWrapper>
  </Fragment>
);

/*------------------------------------------------*/
/* EXPORTS
/*------------------------------------------------*/
export default Layout;
