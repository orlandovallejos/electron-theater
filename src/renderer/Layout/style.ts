/*------------------------------------------------*/
/* LIBRARIES
/*------------------------------------------------*/
import styled from 'styled-components';
/*------------------------------------------------*/
/* INTERNAL DEPENDENCIES
/*------------------------------------------------*/
import theme from '../themes/default';

const LayoutWrapper = styled.div`
  background: red;
  display: flex;
  height: 100vh;
`;
LayoutWrapper.displayName = 'LayoutWrapper';

// const SidebarWrapper = styled.div`
//   background: green;
//   width: 250px;
// `;
// SidebarWrapper.displayName = 'SidebarWrapper';

const ContentWrapper = styled.div`
  background: ${theme.background.primary};
  flex: 1;
`;
ContentWrapper.displayName = 'ContentWrapper';

/*------------------------------------------------*/
/* EXPORTS
/*------------------------------------------------*/
export { ContentWrapper };
export default LayoutWrapper;
