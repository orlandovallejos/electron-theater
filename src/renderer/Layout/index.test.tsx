/*------------------------------------------------*/
/* LIBRARIES
/*------------------------------------------------*/
import React from 'react';
import shallow from 'enzyme/shallow';
/*------------------------------------------------*/
/* INTERNAL DEPENDENCIES
/*------------------------------------------------*/
import Layout from './index';

describe('<Layout />', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<Layout />);

    expect(wrapper.debug()).toMatchSnapshot();
  });
});
