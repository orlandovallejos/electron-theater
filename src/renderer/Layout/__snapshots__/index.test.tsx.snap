// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Layout /> should render correctly 1`] = `
"<LayoutWrapper>
  <ContentWrapper>
    <MemoryRouter>
      <Switch>
        <Route path=\\"/\\" component={[Function: Home]} />
      </Switch>
    </MemoryRouter>
  </ContentWrapper>
</LayoutWrapper>"
`;
